.github/workflows/main.yml

```diff
@@ -132,14 +132,14 @@ jobs:
     - run: git fetch --prune --unshallow
     - name: Set up Python ${{ matrix.python }}
       uses: actions/setup-python@v2
+      if: matrix.python != '3.9-dev'
+      with:
+        python-version: ${{ matrix.python }}
+    - name: install python ${{ matrix.python }}
```

Comment:
Minor nitpick, feel free to ignore:

The name of the standard action is `Set up Python {version}`, but for deadsnakes it is `install python {version}`. Maybe use something similar between them, say `Set up Python {version} (deadsnakes)` for the new action?